utils/terrascript_client.py

```diff
@@ -1726,3 +1735,183 @@ def unpack_namespace_info(self, namespace_info):
         cluster = namespace_info['cluster']['name']
         namespace = namespace_info['name']
         return cluster, namespace
+
+    @staticmethod
+    def validate_elasticsearch_version(version):
+        """ Validate ElasticSearch version. """
+        return version in [7.4, 7.1,
+                           6.8, 6.7, 6.5, 6.4, 6.3, 6.2, 6.0,
+                           5.6, 5.5, 5.3, 5.1,
+                           2.3,
+                           1.5]
+
+    @staticmethod
+    def get_elasticsearch_service_role_tf_resource(identifier):
+        """ Service role for ElasticSearch. """
+        service_role = {
+          'aws_service_name': "es.amazonaws.com"
+        }
+        return aws_iam_service_linked_role(identifier, **service_role)
+
+    @staticmethod
+    def is_elasticsearch_domain_name_valid(name):
+        """ Handle for Error creating Elasticsearch:
+        InvalidParameterValue: Elasticsearch domain name must must start with a
+        lowercase letter and must be between 3 and 28 characters. Valid
+        characters are a-z (lowercase only), 0-9, and - (hyphen). """
+        if len(name) < 3 or len(name) > 28:
+            return False
+        pattern = r'^[a-z][a-z0-9-]+$'
+        return re.search(pattern, name)
+
+    def populate_tf_resource_elasticsearch(self, resource, namespace_info,
+                                           existing_secrets):
+
+        account, identifier, values, output_prefix, output_resource_name = \
+            self.init_values(resource, namespace_info)
+
+        tf_resources = []
+
+        self.init_common_outputs(tf_resources, namespace_info,
+                                 output_prefix, output_resource_name)
+
+        if values['identifier'] and \
```

Comment:
Should just be able to use identifier from the return of init_values.  It also has to exist so should you be checking for len > 0?

Reply:
I don't see any other integrations check for length of identifier

Reply:
If we want to check the length of identifier maybe we should add check in the init_values func? WDYT?

Reply:
I was just thinking in the context of this check.  You're looking for a non-none identifier in values, but it will always exist as a minimum empty string.  Other integrations don't check for len of identifier, but I was suggesting a check that might be worthwhile.  so long as is_elasticsearch_domain_name_valid can handle an empty string, I don't think you need to add a len check.  Removing the first condition here seems like it would be fine.

Reply:
updated